numpyro/distributions/continuous.py

```diff
@@ -501,13 +498,13 @@ def variance(self):
 
 class Pareto(TransformedDistribution):
     arg_constraints = {'alpha': constraints.positive, 'scale': constraints.positive}
-    support = constraints.real
 
     def __init__(self, alpha, scale=1., validate_args=None):
         batch_shape = lax.broadcast_shapes(np.shape(scale), np.shape(alpha))
         self.scale, self.alpha = np.broadcast_to(scale, batch_shape), np.broadcast_to(alpha, batch_shape)
         base_dist = Exponential(self.alpha)
-        transforms = [ExpTransform(), AffineTransform(loc=0, scale=self.scale)]
+        transforms = [ExpTransform(domain=constraints.positive),
```

Comment:
It would be nice if `domain` could also be set later via a setter. Then we wouldn't have to pass domain to any of these transforms and we could just infer it in `TransformedDistribution`, but that can be delegated to another PR perhaps.

Reply:
Yeah, that would be ideal. This kind of pattern should be the default behaviour of `TransformedDistribution` (and `ComposedTransformation`), where `domain` is the support of base distribution and codomain is retrieved recursively. This will prevent errors when users try to use TransformedDistribution but forget to redefine a correct support. By using setter, we can also take care of the domain/codomain relation for `transform.inv`. We can override the default behaviour if we already know the support. I'll refine the default behaviour and defer the `setter` pattern to a later PR, when the idea is more consolidated. Will add a TODO for it too.